apps/email/js/html_cache_restore.js

```diff
@@ -21,23 +66,143 @@ var startupCacheEventsSent = false;
  * Set by build process. Set as a global because it
  * is also used in html_cache.js.
  */
-var HTML_COOKIE_CACHE_VERSION = '2';
+var HTML_CACHE_VERSION = '2';
 
 /**
- * Max size of cookie cache segments. Set as a global because it is also used by
- * html_cache.js.
+ * A function callback is registered if the model needs to be loaded before
+ * final view determination can be done. This should be a rare event, but can
+ * happen, see localOnModelLoaded for more details. config.js looks for this
+ * callback.
  */
-var HTML_COOKIE_CACHE_MAX_SEGMENTS = 40;
-
-// Use a global to work around issue with
-// navigator.mozHasPendingMessage only returning
-// true to the first call made to it.
-window.htmlCacheRestorePendingMessage = [];
+var startupOnModelLoaded;
```

Comment:
I think it would be helpful for this to be a bit more blatant about the fact that it's a global.  I tend to like explicitly doing "window.strtupOnModelLoaded" (probably because it's both obvious and shuts up js2-mode's syntax highlighting since it expects window to be a global), but other things could work.  At the very least, I think the doc-block should say the word global.